src/java.base/share/classes/java/time/InstantSource.java

```diff
@@ -0,0 +1,243 @@
+    }
+
+    //-------------------------------------------------------------------------
+    /**
+     * Obtains a source that returns instants from the specified source truncated to
+     * the nearest occurrence of the specified duration.
+     * <p>
+     * This source will only tick as per the specified duration. Thus, if the
+     * duration is half a second, the source will return instants truncated to the
+     * half second.
+     * <p>
+     * The tick duration must be positive. If it has a part smaller than a whole
+     * millisecond, then the whole duration must divide into one second without
+     * leaving a remainder. All normal tick durations will match these criteria,
+     * including any multiple of hours, minutes, seconds and milliseconds, and
+     * sensible nanosecond durations, such as 20ns, 250,000ns and 500,000ns.
+     * <p>
+     * A duration of zero or one nanosecond would have no truncation effect. Passing
+     * one of these will return the underlying source.
+     * <p>
+     * Implementations may use a caching strategy for performance reasons. As such,
+     * it is possible that the start of the requested duration observed via this
+     * source will be later than that observed directly via the underlying source.
+     * <p>
+     * The returned implementation is immutable, thread-safe and
+     * {@code Serializable} providing that the base source is.
+     *
+     * @param baseSource  the base source to base the ticking source on, not null
+     * @param tickDuration  the duration of each visible tick, not negative, not null
+     * @return a source that ticks in whole units of the duration, not null
+     * @throws IllegalArgumentException if the duration is negative, or has a
+     *  part smaller than a whole millisecond such that the whole duration is not
+     *  divisible into one second
+     * @throws ArithmeticException if the duration is too large to be represented as nanos
+     */
+    static InstantSource tick(InstantSource baseSource, Duration tickDuration) {
+        Objects.requireNonNull(baseSource, "baseSource");
+        return Clock.tick(baseSource.withZone(ZoneOffset.UTC), tickDuration);
+    }
+
+    //-----------------------------------------------------------------------
+    /**
+     * Obtains a source that always returns the same instant.
+     * <p>
+     * This source simply returns the specified instant.
+     * As such, it is not a source that represents the current instant.
+     * The main use case for this is in testing, where the fixed source ensures
+     * tests are not dependent on the current source.
+     * <p>
+     * The returned implementation is immutable, thread-safe and {@code Serializable}.
+     *
+     * @param fixedInstant  the instant to use, not null
+     * @return a source that always returns the same instant, not null
+     */
+    static InstantSource fixed(Instant fixedInstant) {
+        return Clock.fixed(fixedInstant, ZoneOffset.UTC);
```

Comment:
Instant might also implement InstantSource, returning itself.
This fixed method would be unnecessary because an Instant is itself a InstantSource.

Reply:
While I can see the appeal, I don't think this would be a good choice. Doing so would add three methods to `Instant` that are of no value. `millis()` would duplicate `toEpochMilli()`, `withZone(zone)` would duplicate `atZone(zone)` and `instant()` would return `this`. Moreover it doesn't respect the rationale of the interface, which is explictly to return the current instant, not any random instant.

Effectively, this proposal is a variation of Scala's implicit conversions which were rejected during the development of JSR-310 as being more likely to cause bugs than help developers,